examples/partial_weight_exchange_example/client.py

```diff
@@ -88,56 +79,12 @@ def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
         self.num_examples = num_examples
         return train_loader, val_loader
 
-    def get_parameter_exchanger(self, config: Config) -> ParameterExchanger:
-        normalize = self.narrow_config_type(config, "normalize", bool)
-        filter_by_percentage = self.narrow_config_type(config, "filter_by_percentage", bool)
-        norm_threshold = self.narrow_config_type(config, "norm_threshold", float)
-        exchange_percentage = self.narrow_config_type(config, "exchange_percentage", float)
-        parameter_exchanger = NormDriftParameterExchanger(
-            norm_threshold, exchange_percentage, normalize, filter_by_percentage
-        )
-        return parameter_exchanger
-
     def get_criterion(self, config: Config) -> _Loss:
         return torch.nn.CrossEntropyLoss()
 
     def get_optimizer(self, config: Config) -> Optimizer:
         return torch.optim.AdamW(self.model.parameters(), lr=0.00001, weight_decay=0.001)
 
-    def get_parameters(self, config: Config) -> NDArrays:
-        # Determines which weights are sent back to the server for aggregation. This uses a parameter exchanger to
-        # determine parameters sent
-        assert self.model is not None and self.parameter_exchanger is not None
-        return self.parameter_exchanger.push_parameters(self.model, self.initial_model, config=config)
-
-    def set_parameters(self, parameters: NDArrays, config: Config) -> None:
-        """
-        Sets the local model parameters transfered from the server using a parameter exchanger to coordinate how
-        parameters are set. If it's the first time the model is being initialized, we assume the full model is being
-        initialized and the weights sent correspond to the complete set of weights. Thus we use the
-        FullParameterExchanger() to set all model weights.
-        Subsequently, this approach uses the threshold parameter exchanger to handle exchanging a dynamic subset of
-        model layer weights.
-        Args:
-            parameters (NDArrays): In the first setting, parameters represents the full set of weights for the model
-            for initialization. Thereafter, we assume that it is a subset of the weights packed alongside the names
-            of the layer to which the weights correspond.
-            config (Config): The config is sent by the FL server to allow for customization in the function if desired.
-        """
-        if not self.model_weights_initialized:
-            self.initialize_all_model_weights(parameters, config)
-        else:
-            self.parameter_exchanger.pull_parameters(parameters, self.model, config)
-        # stores the values of the new model parameters at the beginning of each training round.
-        self._align_model_parameters(self.initial_model, self.model)
-
-    def _align_model_parameters(self, initial_model: nn.Module, target_model: nn.Module) -> None:
```

Comment:
@yc7z: I feel like, at some point, we had a good reason for this function, but I couldn't quite see why we needed it anymore. It was only called on line 132 above and it appears that it was setting the `initial_model` values before training to use as a comparison to the trained model after local training. However, if initial_model is the same architecture as self.model, which is set using the server passed weights in `set_parameters`, I don't see why we need the for loops in this function. So I put
```self.initial_model.load_state_dict(self.model.state_dict(), strict=True)```
at the end of `set_parameters` in the new refactor. Let me know if that makes sense to you or if I'm missing something.

Reply:
Yes, this makes sense. Originally we had this method because I was using `self.pull_parameters()` to update `self.initial_model`, which was incorrect since the parameters pulled would not be the full model weights. But I think we could have simply used `load_state_dict` instead back then.